src/mono/mono/mini/mini-posix.c

```diff
@@ -517,48 +518,11 @@ static volatile gint32 sampling_thread_running;
 
 #ifdef HOST_DARWIN
 
-static clock_serv_t sampling_clock_service;
+static clock_serv_t sampling_clock;
 
-static void
```

Comment:
This would need to stay static if we're going this route, but I think it's better to scrap the function on non-Linux and put it behind an ifdef at the call site.

Reply:
Oops, good catch!